streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -144,6 +144,13 @@ class StreamingContext private[streaming] (
     }
   }
 
+  if (sc.conf.contains("spark.cores.max")) {
+    val totalCores = sc.conf.getInt("spark.cores.max", 1)
```

Comment:
The config `spark.cores.max` is used to limit the max number of cores that a single executor can require, and as @jerryshao pointed out, it's not set by default. I'm not convinced that validating the config can benefit your issue.